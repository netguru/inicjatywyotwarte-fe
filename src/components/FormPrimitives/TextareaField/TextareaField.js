import React from 'react'
import styled from 'styled-components'
import Label from 'components/FormPrimitives/FieldLabel/FieldLabel'
import Error from 'components/FormPrimitives/FieldError/FieldError'

const InputBox = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5rem;
  width: 100%;
`

const Textarea = styled.textarea`
  min-height: 8rem;
  border: 1px solid ${props =>
    props.invalid
    ? props.theme.warningColor
    : props.theme.modalColorVeryLight
  };
  border-radius: ${props => props.theme.borderRadiusSmall};
  font-size: 0.9rem;
  padding: 0.8rem 0.8rem;
  font-weight: 100;
  outline: none;
  resize: vertical;
  color: ${props => props.theme.modalColorLight};
  background: ${props => props.theme.lightBackgroundColor};

  &:focus,
  &:active {
    border: ${props => props.theme.inputActiveBorder};
  }
`

export default function TextareaField ({
  name,
  onChange,
  error,
  label
}) {
  return (
    <InputBox>
      <Label htmlFor={name}>{label}</Label>
      <Textarea
        id={name}
        invalid={Boolean(error)}
        onInput={() => onChange()}
      />
      {error && <Error>{error}</Error>}
    </InputBox>
  )
}
