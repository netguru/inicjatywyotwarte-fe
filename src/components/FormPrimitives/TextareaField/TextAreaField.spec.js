import React from 'react'
import TextareaField from './TextareaField'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const baseProps = {
  name: 'name',
  onChange: jest.fn(),
  label: 'label',
  error: null
}

describe('TextareaField', () => {
  test('renders textarea field', () => {
    const wrapper = shallow(<TextareaField {...baseProps} />);
    expect(wrapper.find('InputBox').length).toEqual(1);
  });

  test('renders textarea field with error', () => {
    const propsWithError = {
      ...baseProps,
      error: 'error text message'
    };
    const wrapper = shallow(<TextareaField {...propsWithError} />);
    expect(wrapper.find('[className="error"]').length).toEqual(1);
  });

  test('match snapshot', () => {
    const wrapper = shallow(<TextareaField {...baseProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


