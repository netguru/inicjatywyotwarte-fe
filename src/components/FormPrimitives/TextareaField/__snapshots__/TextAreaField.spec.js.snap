// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaField match snapshot 1`] = `
<InputBox>
  <styled.label
    htmlFor="name"
  >
    label
  </styled.label>
  <Textarea
    id="name"
    invalid={false}
    onInput={[Function]}
  />
</InputBox>
`;
